cpp/src/parquet/encoding.cc

```diff
@@ -3382,6 +3385,17 @@ class DeltaByteArrayDecoderImpl : public DecoderImpl, virtual public TypedDecode
       if (ARROW_PREDICT_FALSE(static_cast<size_t>(prefix_len_ptr[i]) > prefix.length())) {
         throw ParquetException("prefix length too large in DELTA_BYTE_ARRAY");
       }
+      if (prefix_len_ptr[i] == 0) {
```

Comment:
Maybe I'm missing something, but it seems that if we had `prefix_len_ptr[i] == 0` but `buffer[i].len != 0` several times in a row, we'd skip copying some buffers in. For example, in the case where the encoded array is `["aaa", "bbb", "ccc", "cccddd"]` the prefix length would be 0 for the loop iteration corresponding to "bbb" and we'd assign it to `prefix` and skip to the next iteration... where the prefix length would still be 0 and we'd assign "ccc" to `prefix` without copying anything

Reply:
Emmm I don't catch what you mean. When meet the case you're saying, when handling "cccddd":

1. buffer -> point to "ddd", length = 3
2. prefix = "ccc". prefix-length = 3

So here neither case is meet. I think `prefix` doesn't means `prefix`, actually it just means `last-value`

Reply:
You can refer to here for detail: https://github.com/apache/parquet-format/blob/master/Encodings.md#delta-strings-delta_byte_array--7